### ISSUE TITLE
[A11y] Supports links inside ListTile

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#131619]
b/245911732

### Steps to reproduce

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MaterialApp(
    home: Scaffold(
      body: Center(child: CheckboxListTile(
        value: false,
        onChanged: (_) {},
        title: Text.rich(TextSpan(
          children: [
            TextSpan(text:'before'),
            WidgetSpan(child: TextButton(child: Text('link'), onPressed: (){}, )),
            TextSpan(text:'after'),
          ]
        )),
      )),
    )
  ));
}

```

1. run the app with talkback or voiceover.

### Expected results

The link should have its own semantics node.

### Actual results

The entire checkboxlisttile is merged into one node, and the link widget is always at the end of the label.

### Logs

<details>
<summary>Logs</summary>

```console

<!-- Paste your logs here -->

```

</details>

### Flutter Doctor output

<details>
<summary>Doctor output</summary>

```console

<!-- Paste your output here -->

```

</details>

**Comment by QuncCccccc** [comment]
Will be out of office for about one month. Here's an update (just in case anyone will handle this issue next):

Tried to replace the `MergeSemantics` in `CheckboxListTile` with `Semantics(container: true)` but the semantic flags in `Checkbox` cannot merge into `CheckboxListTile` Semantics node and will break the CheckboxListTile test in `dev/a11y_assessments/test/accessibility_guideline_test.dart`.(draft pr: #131709).

<details>
<summary>Code sample</summary>

```dart

import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() => runApp(const CheckboxListTileApp());

class CheckboxListTileApp extends StatelessWidget {
  const CheckboxListTileApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(useMaterial3: true),
      home: const CheckboxListTileExample(),
    );
  }
}

class CheckboxListTileExample extends StatefulWidget {
  const CheckboxListTileExample({super.key});

  @override
  State<CheckboxListTileExample> createState() => _CheckboxListTileExampleState();
}

class _CheckboxListTileExampleState extends State<CheckboxListTileExample> {
  bool checkboxValue1 = true;

  @override
  Widget build(BuildContext context) {
    debugDumpSemanticsTree();
    return Scaffold(
      appBar: AppBar(title: const Text('CheckboxListTile Sample')),
      body: Column(
        children: <Widget>[
          CheckboxListTile(
            value: checkboxValue1,
            onChanged: (bool? value) {
              setState(() {
                checkboxValue1 = value!;
              });
            },
            title: Text.rich(TextSpan(
                children: <InlineSpan>[
                  const TextSpan(text: 'before'),
                  WidgetSpan(child: TextButton(child: const Text('link'), onPressed: (){},)),
                  const TextSpan(text: 'after'),
                ]
            )),
            subtitle: const Text('Supporting text'),
          ),
        ],
      ),
    );
  }
}

```
</details>

<details>
<summary>Semantics tree</summary>

flutter: SemanticsNode#0
flutter:  │ Rect.fromLTRB(0.0, 0.0, 1600.0, 1200.0)
flutter:  │
flutter:  └─SemanticsNode#1
flutter:    │ Rect.fromLTRB(0.0, 0.0, 800.0, 600.0) scaled by 2.0x
flutter:    │ textDirection: ltr
flutter:    │
flutter:    └─SemanticsNode#2
flutter:      │ Rect.fromLTRB(0.0, 0.0, 800.0, 600.0)
flutter:      │ sortKey: OrdinalSortKey#d53cd(order: 0.0)
flutter:      │
flutter:      └─SemanticsNode#3
flutter:        │ Rect.fromLTRB(0.0, 0.0, 800.0, 600.0)
flutter:        │ flags: scopesRoute
flutter:        │
flutter:        ├─SemanticsNode#7
flutter:        │ │ Rect.fromLTRB(0.0, 0.0, 800.0, 56.0)
flutter:        │ │
flutter:        │ └─SemanticsNode#8
flutter:        │     Rect.fromLTRB(270.0, 14.0, 530.0, 42.0)
flutter:        │     flags: isHeader
flutter:        │     label: "CheckboxListTile Sample"
flutter:        │     textDirection: ltr
flutter:        │
flutter:        └─SemanticsNode#4
flutter:          │ Rect.fromLTRB(0.0, 56.0, 800.0, 127.0)
flutter:          │ actions: tap
flutter:          │ flags: hasEnabledState, isEnabled, isFocusable
flutter:          │ label:
flutter:          │   "before
flutter:          │   after
flutter:          │   Supporting text"
flutter:          │ textDirection: ltr
flutter:          │
flutter:          ├─SemanticsNode#5
flutter:          │   Rect.fromLTRB(68.1, 8.4, 124.1, 40.4)
flutter:          │   tags: PlaceholderSpanIndexSemanticsTag(0)
flutter:          │   actions: tap
flutter:          │   flags: isButton, hasEnabledState, isEnabled, isFocusable
flutter:          │   label: "link"
flutter:          │   textDirection: ltr
flutter:          │
flutter:          └─SemanticsNode#6
flutter:              Rect.fromLTRB(744.0, 19.5, 776.0, 51.5)
flutter:              actions: tap
flutter:              flags: hasCheckedState, isChecked, hasEnabledState, isEnabled,
flutter:                isFocusable
flutter: 

</details>

For the next step, we need to investigate why the checkbox semantics node doesn't merge to the CheckboxListTile's semantics node. One reason that we thought was the conflict between `ListTile`'s `onTap` and `Checkbox`'s `onTap`, but even I assigned a `null` to one of the `onTap`, the Checkbox semantics flags still cannot be merged up.

The other direction that we can try is to try to add an override method `describeSemanticsConfiguration` method in the [custom renderer](https://github.com/flutter/flutter/blob/0bc5a2bca440ba853629ee575e59700b39de2ab6/packages/flutter/lib/src/material/list_tile.dart#L1052) class in `ListTile`, then merge leading, title, subtitle and trailing. But not sure if this can fix the problem.

CC: @chunhtai

**Comment by niteshsh4rma** [comment]
@QuncCccccc are you guys working on this?
